src/erlcloud_ec2.erl

```diff
@@ -1713,43 +1786,81 @@ extract_reserved_instance(Node) ->
      {state, get_text("state", Node)}
     ].
 
-%%
-%%
--spec describe_reserved_instances_offerings() -> proplist().
-describe_reserved_instances_offerings() -> describe_reserved_instances_offerings([]).
+-type ec2_selector() :: proplist().
+
+-define(RESERVED_INSTANCES_OFFERINGS_MR_MIN, 5).
+-define(RESERVED_INSTANCES_OFFERINGS_MR_MAX, 1000).
 
--spec describe_reserved_instances_offerings([{atom(), string()}] | aws_config()) -> proplist().
+-spec describe_reserved_instances_offerings() -> ok_error(proplist()).
+describe_reserved_instances_offerings() ->
+    describe_reserved_instances_offerings([], default_config()).
+
+-spec describe_reserved_instances_offerings(ec2_selector()) -> ok_error(proplist());
+(aws_config()) -> ok_error(proplist()).
+describe_reserved_instances_offerings(Selector)
+    when is_list(Selector) ->
+    describe_reserved_instances_offerings(Selector, default_config());
 describe_reserved_instances_offerings(Config)
-  when is_record(Config, aws_config) ->
-    describe_reserved_instances_offerings([], Config);
-describe_reserved_instances_offerings(Selector) ->
-    describe_reserved_instances_offerings(Selector, default_config()).
+    when is_record(Config, aws_config) ->
+    describe_reserved_instances_offerings([], Config).
 
--spec describe_reserved_instances_offerings([{atom(), string()}], aws_config()) -> proplist().
+-spec describe_reserved_instances_offerings(ec2_selector(), aws_config())
+    -> ok_error(proplist()) | ok_error(proplist(), ec2_token()).
 describe_reserved_instances_offerings(Selector, Config)
-  when is_list(Selector) ->
-    InstanceTypes = [Value || {Key, Value} <- Selector, Key =:= instance_type],
-    AvailabilityZones = [Value || {Key, Value} <- Selector, Key =:= availability_zone],
-    Descs = [Value || {Key, Value} <- Selector, Key =:= product_description],
-    Token = [Value || {Key, Value} <- Selector, Key =:= token],
+    when is_list(Selector), is_record(Config, aws_config) ->
+    InstanceTypes = proplists:get_value(instance_type, Selector),
```

Comment:
technically speaking previous implementation allowed several `instance_type`s  or rest of the filters to be passed.
Please check with AWS API doc and make sure it's implemented accordingly